src/common/src/session_config/non_zero64.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::num::NonZeroU64;
+use std::str::FromStr;
+
+#[derive(Copy, Default, Debug, Clone, PartialEq, Eq)]
+pub struct ConfigNonZeroU64(pub Option<NonZeroU64>);
+
+impl FromStr for ConfigNonZeroU64 {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parsed = s.parse::<u64>().map_err(|_| ())?;
+        if parsed == 0 {
+            Ok(Self(None))
```

Comment:
It is confusing that this `NonZero` type accepts `0` value. Seems that `0` input means default configuration? Could you clarify it in comments?

Reply:
Can you suggest a better name?

Reply:
```
/// When set this config as `0`, the value is `None`, otherwise the value is
/// `Some(val)`
```

The naming is okay to me.

But does this behavior apply to **all** non-zero parameters? It looks a bit tricky to me.

Personally I tend to consider this as an behaivor of specific parameter.